example/reader_util.c

```diff
@@ -1745,6 +1745,29 @@ int ndpi_is_datalink_supported(int datalink_type) {
   }
 }
 
+static bool ndpi_is_valid_vxlan(const struct pcap_pkthdr *header, const u_char *packet, u_int16_t ip_offset, u_int16_t ip_len){
+  if(header->caplen >= ip_offset + ip_len + sizeof(struct ndpi_vxlanhdr)) {
```

Comment:
To simplify the code a bit,
```suggestion
  if(header->caplen < ip_offset + ip_len + sizeof(struct ndpi_vxlanhdr)) {
    return false;
  }
```
may also be an option.